### PR TITLE
Push connectivity-related events through DisplaySystemMessage

### PR DESCRIPTION
**Pull request by ChrisKitching**
According to the "discussion" over in #276, these count as system messages, too. The likely cause of that bug is that those messages were just being printed with AddChatText instead of being subjected to something that checks SystemMessagesEnabled.
It seems that DisplaySystemMessage is the "Display a system message if it's enabled" function. Let's push the problematic messages through it so they'll be muffled if the tickbox is set.

There was also logic in DisplaySystemMessage that causes some messages to be displayed irrespective of the setting. Either the nomenclature, or this behaviour, needs to be changed. It's either a "system message" (in which case it should be shown only if those are enabled), or it's some other sort of message that we just display. It's very strange for a function called DisplaySystemMessage to in some cases show a message given that SystemMessagesEnabled() is false.

To sum up:

- The cause of #276 is that the problematic messages have never been printed in a way that checks SystemMessagesEnabled.
- We need to consider precisely what we mean by "system message" and decide what, if any, messages should be exempt from filtering. Do we even want this feature at all? It seems like you definitely want to hear about players being disconnected from your lobby...

### CODE REVIEW

**Comment by ChrisKitching** [comment]
... Aaaand this is why we don't write code at 0300.
